src/cleanfigure.m

```diff
@@ -786,57 +718,59 @@ function movePointsCloser(meta, handle)
 end
 % =========================================================================
 function xNew = moveToBox(x, xRef, xLim, yLim)
-  % Takes a box defined by xlim, ylim, one point x and a reference point
-  % xRef.
-  % Returns the point xNew that sits on the line segment between x and xRef
-  % *and* on the box. If several such points exist, take the closest one
-  % to x.
+  % Takes a box defined by xlim, ylim, a vector of points x and a vector of 
+  % reference points xRef.
+  % Returns the vector of points xNew that sits on the line segment between 
+  % x and xRef *and* on the box. If several such points exist, take the 
+  % closest one to x.
 
   % Find out with which border the line x---xRef intersects, and determine
   % the smallest parameter alpha such that x + alpha*(xRef-x)
   % sits on the boundary.
-  minAlpha = inf;
+  n = size(x, 1);
+  minAlpha = inf(n, 1);
+  
+  % Get the corner points
   [bottomLeft, topLeft, bottomRight, topRight] = corners(xLim, yLim);
-  %TODO: clean up duplicate code below, possibly store lambda in matrix
 
   % left boundary:
-  lambda = crossLines(x, xRef, bottomLeft, topLeft);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
+  minAlpha = UpdateAlpha(x, xRef, bottomLeft, topLeft, minAlpha);
 
   % bottom boundary:
-  lambda = crossLines(x, xRef, bottomLeft, bottomRight);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
+  minAlpha = UpdateAlpha(x, xRef, bottomLeft, bottomRight, minAlpha);
 
   % right boundary:
-  lambda = crossLines(x, xRef, bottomRight, topRight);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
-
+  minAlpha = UpdateAlpha(x, xRef, bottomRight, topRight, minAlpha);
+  
   % top boundary:
-  lambda = crossLines(x, xRef, topLeft, topRight);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
+  minAlpha = UpdateAlpha(x, xRef, topLeft, topRight, minAlpha);
 
-  % create the new point
-  xNew = x + minAlpha*(xRef-x);
+  % Create the new point
+  xNew = x + bsxfun(@times ,minAlpha, (xRef-x));
+end
+% =========================================================================
+function minAlpha = UpdateAlpha(X1, X2, X3, X4, minAlpha)
```

Comment:
Next to the stylistic remark of @okomarov, I think it's not clear what this function actually does. It's called `UpdateAlpha`, it's H1 line states that it checks for an intersection of the segments `X1--X2` and `X3--X4`.
What isn't stated is what this `alpha` is and why it should be updated: that's something that's only mentioned in its caller.

It might be useful to decouple the intersection calculation and minimization of `alpha`. That might pass around bigger data vectors, I suppose, but it keeps everything more readable.